// gRPC Protocol Buffer tanımı - iPhone Fiyat Tahmini

syntax = "proto3";

package iphone_price_prediction;

// Fiyat tahmin servisi
service PricePrediction {
    // Fiyat tahmini yap
    rpc PredictPrice(PhoneSpec) returns (PriceResponse);
    
    // Model bilgilerini getir
    rpc GetModelInfo(ModelInfoRequest) returns (ModelInfoResponse);
    
    // Health check
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

// Telefon özellikleri (Input)
message PhoneSpec {
    int32 model_id = 1;
    int32 ram_gb = 2;
    int32 storage_gb = 3;
    string condition = 4;  // Mükemmel, Çok İyi, İyi, Orta
    int32 release_year = 5;
}

// Fiyat tahmini sonucu (Output)
message PriceResponse {
    double predicted_price = 1;
    double confidence_score = 2;
    PriceRange price_range = 3;
    string status = 4;
    string message = 5;
}

// Fiyat aralığı
message PriceRange {
    double min_price = 1;
    double max_price = 2;
}

// Model bilgi isteği
message ModelInfoRequest {
    int32 model_id = 1;
}

// Model bilgi yanıtı
message ModelInfoResponse {
    string model_name = 1;
    int32 release_year = 2;
    repeated int32 available_storage = 3;
    int32 ram_gb = 4;
    bool is_pro = 5;
}

// Health check isteği
message HealthCheckRequest {
    string service = 1;
}

// Health check yanıtı
message HealthCheckResponse {
    string status = 1;
    string version = 2;
    bool model_loaded = 3;
    string uptime = 4;
}

